src/main/lombok/org/heigit/ohsome/ohsomeapi/executor/ContributionsExecutor.java

```diff
@@ -66,13 +95,98 @@ public Response count(boolean isUsersRequest) throws Exception {
+  }
+  
+  /**
+   * Performs a count calculation for /contributions/count, /contributions/density or
+   * /contribution/latest/count.
+   * 
+   * @param mapRed a MapReducer of OSM contributions
+   * @param isContributionsLatest the boolean value relative to the endpoint /contributions/latest 
+   * @return SortedMap with counts of contributions aggregated by timestamp
+   * @throws Exception thrown by {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapAggregator
+   *         #count() count}
+   * @throws UnsupportedOperationException thrown by
+   *         {@link org.heigit.bigspatialdata.oshdb.api.mapreducer.MapReducer#aggregateByTimestamp()
+   *         aggregateByTimeStamp}
+   */
+  private SortedMap<OSHDBTimestamp, Integer> contributionsCount(MapReducer<OSMContribution> mapRed,
+      boolean isContributionsLatest) throws UnsupportedOperationException, Exception {
+    if (isContributionsLatest) {
+      MapReducer<List<OSMContribution>> mapRedGroupByEntity = mapRed.groupByEntity();
+      Optional<FilterExpression> filter = processingData.getFilterExpression();
+      if (filter.isPresent()) {
+        mapRedGroupByEntity = mapRedGroupByEntity.filter(filter.get());
+      }
+      return contributionsFilter(mapRedGroupByEntity
+          .map(listContrsPerEntity -> listContrsPerEntity.get(listContrsPerEntity.size() - 1)))
+          .aggregateByTimestamp(OSMContribution::getTimestamp).count();
+    } else {
+      return contributionsFilter(mapRed).aggregateByTimestamp().count();
+    }
+  }
 
+  /**
+   * Filters contributions by contribution type.
+   * 
+   * @param mapRed a MapReducer to be filtered
+   * @return MapReducer filtered by contribution type
+   */
+  private MapReducer<OSMContribution> contributionsFilter(MapReducer<OSMContribution> mapRed) {
+    String types = servletRequest.getParameter("contributionType");
+    if (types == null) {
+      return mapRed;
+    }
+    types = types.toUpperCase();
+    List<String> givenTypes = Arrays.asList(types.split(","));
+    for (int i = 0; i < givenTypes.size(); i++) {
+      switch (givenTypes.get(i)) {
+        case "TAGCHANGE":
+          givenTypes.set(i, "TAG_CHANGE");
+          break;
+        case "GEOMETRYCHANGE":
+          givenTypes.set(i, "GEOMETRY_CHANGE");
+          break;
+        case "CREATION":
+          break;
+        case "DELETION":
+          break;
+        default:
+          throw new BadRequestException("The contribution type must be 'creation', 'deletion',"
+              + "'geometryChange', 'tagChange' or a combination of them");
+      }
+    }
+    List<ContributionType> contributionTypes = new ArrayList<>();
+    givenTypes.stream().forEach(givenType -> {
+      for (ContributionType contrType : ContributionType.values()) {
+        if (contrType.name().equals(givenType)) {
+          contributionTypes.add(contrType);
+        }
+      }
+    });
```

Comment:
I strongly advice to use static typing wherever possible. Here you are circumventing the compiler's ability to check types (specifically the `ContributionType` enum) by looping through the enum values and comparing them to strings. IMHO this does not provide any benefit over something simple like this:

```java
    List<String> contributionTypes = new ArrayList<>();
    for (String givenType : types.split(",")) {
      switch (givenType) {
        case "CREATION":
          contributionTypes.add(ContributionType.CREATION);
          break;
        case "DELETION":
          contributionTypes.add(ContributionType.DELETION);
          break;
        case "TAGCHANGE":
          contributionTypes.add(ContributionType.TAG_CHANGE);
          break;
        case "GEOMETRYCHANGE":
          contributionTypes.add(ContributionType.GEOMETRY_CHANGE);
          break;
        default:
          throw new BadRequestException("The contribution type must be 'creation', 'deletion',"
              + "'geometryChange', 'tagChange' or a combination of them");
      }
    }
```